### DISCUSSION TITLE
Does the pyhealth.synthetic.halo exist? Has HALO been pulled from PyHealth?

### DISCUSSION THREAD

**Discussion by mirzafarhan7** [sunlabuiuc/PyHealth#278] [Q&A]
Does the pyhealth.synthetic.halo exist? Has HALO been pulled from PyHealth?

**Comment by userjuma** [comment]
HALO is available in PyHealth (located at `pyhealth/models/generators/halo.py`, verified on `master` for PyHealth 2.0.1).It is exported directly from `pyhealth.models`:
python
from pyhealth.models import HALO

model = HALO(
    dataset=your_dataset,
    embed_dim=16,
    n_heads=2,
    n_layers=2,
    n_ctx=8,
    batch_size=2,
    epochs=1,
)
Note that `embed_dim` must be divisible by `n_heads`.
For a complete runnable setup, refer to the test case in `tests/core/test_halo.py`.
